src/runtime/internal/style_manager.ts

```diff
@@ -53,14 +59,19 @@ export function delete_rule(node: Element & ElementCSSInlineStyle, name?: string
 		)
 		.join(', ');
 
-	if (name && !--active) clear_rules();
+	active = Math.max(0, active - 1);
```

Comment:
This feels like it's masking a bug — is the behaviour different when iframes are involved, or is this unrelated to iframes?

Reply:
Yes, it _is_ masking a bug. Toggling a transition on and off multiple times has a memory leak.

<img width="1321" alt="Screen Shot 2019-11-19 at 4 37 43 PM" src="https://user-images.githubusercontent.com/120294/69196159-ea6f2900-0aea-11ea-85f7-ad0c72bb78c6.png">